core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -93,9 +102,17 @@ trait CassandraRecovery extends CassandraTagRecovery with TaggedPreparedStatemen
           extractor = Extractors.persistentRepr(eventDeserializer, serialization),
           // run the query on the journal dispatcher (not the queries dispatcher)
           dispatcher = sessionSettings.pluginDispatcher)
-        .map(p => queries.mapEvent(p.persistentRepr))
-        .runForeach(replayCallback)
-        .map(_ => ())
+        .map { p =>
+          println(s"# asyncReplayMessages mapEvent ${Thread.currentThread().getName}") // FIXME
+          queries.mapEvent(p.persistentRepr)
+        }
+        .map { p =>
+          println(s"# asyncReplayMessages replayCallback ${Thread.currentThread().getName}") // FIXME
+          replayCallback(p)
+        }
+        .withAttributes(ActorAttributes.dispatcher(sessionSettings.pluginDispatcher))
```

Comment:
We want the Sink.ignore on the same dispatcher as well, so `.to(Sink.ignore)` and then the dispatcher attribute for the whole graph

Reply:
ah, thanks